### PR TITLE
Test new SSE reconnect behavior

### PR DESCRIPTION
**Pull request by andymc12**
The JAX-RS 2.1 spec is being clarified with regard to how the SSE client should reconnect to a remote service.  The details are available in JAX-RS issue [606](
https://github.com/eclipse-ee4j/jaxrs-api/issues/606).  The basic idea is that when there is a non-recoverable error (basically any response code with a status other than 200, 204 or 503 - note that a valid Retry-After header must be set with the 503, and that a Content-Type of text/event-stream must be set for the 200), the error handler should be invoked and so should the completion handler.  On a 204 (no content), only the completion handler should be invoked.

For a 503 with a valid Retry-After header, the client should attempt to reconnect after the appropriate delay (unless the client has closed the SseEventSource prior to reconnecting).

This PR includes the runtime changes to meet the new clarifications on the spec, and includes new test cases for these scenarios. It also includes a Java 2 security fix.

### CODE REVIEW

**Comment by andymc12** [comment]
#build

**Comment by andymc12** [comment]
Single failure is unrelated to these changes.

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.sse.3.2/src/org/apache/cxf/jaxrs/sse/client/SseEventSourceImpl.java [hunk 1]
```diff
@@ -223,44 +229,80 @@ private void connect(String lastEventId) {
...
+                    
                 }
             }
 
+            int status = response.getStatus();
+            String contentType = response.getHeaderString(HttpHeaders.CONTENT_TYPE);
+            if (status != 200 || !MediaType.SERVER_SENT_EVENTS.equals(contentType)) {
+                if (LOG.isLoggable(Level.FINEST)) {
+                    LOG.log(Level.FINEST, "Received " + status + " Content-Type=" + contentType);
+                }
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.sse.3.2/src/org/apache/cxf/jaxrs/sse/client/SseEventSourceImpl.java]: Do you want this log message to indicate that something is not right with the content-Type?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.sse.3.2/src/org/apache/cxf/jaxrs/sse/client/SseEventSourceImpl.java [hunk 2]
```diff
@@ -223,44 +229,80 @@ private void connect(String lastEventId) {
...
-            
+
             if (!state.compareAndSet(SseSourceState.CONNECTING, SseSourceState.OPEN)) {
                 throw new IllegalStateException("The SseEventSource is already in " + state.get() + " state");
             }
-            
+
             LOG.fine("Successfuly opened SSE connection to " + target.getUri());
         } catch (final Exception ex) {
+            LOG.log(Level.WARNING, "ANDY caught exception in connect(...)", ex); // TODO REMOVE THIS!
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.sse.3.2/src/org/apache/cxf/jaxrs/sse/client/SseEventSourceImpl.java]: I assume you are ok with delivering this with "Andy" in it beyond the TODO eye catcher?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.sse.3.2/src/org/apache/cxf/jaxrs/sse/client/SseEventSourceImpl.java [hunk 3]
```diff
@@ -337,6 +378,7 @@ public boolean isOpen() {
 
     @Override
     public boolean close(long timeout, TimeUnit tunit) {
+        LOG.log(Level.WARNING, "ANDY close(...)", new Throwable()); // TODO REMOVE THIS!
```
Comment (jim-krueger) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.sse.3.2/src/org/apache/cxf/jaxrs/sse/client/SseEventSourceImpl.java]: Remove "Andy" maybe?

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.sse.3.2/src/org/apache/cxf/jaxrs/sse/client/SseEventSourceImpl.java (hunk 1 above)
Comment (andymc12) [reply dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.sse.3.2/src/org/apache/cxf/jaxrs/sse/client/SseEventSourceImpl.java]: For this, no - the different WebApplicationException that gets thrown below will also be logged, so it should be apparent why this is an error.

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.sse.3.2/src/org/apache/cxf/jaxrs/sse/client/SseEventSourceImpl.java (hunk 2 above)
Comment (andymc12) [reply dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.sse.3.2/src/org/apache/cxf/jaxrs/sse/client/SseEventSourceImpl.java]: good catch - I'll remove this - and the other occurrences.  :)

**Comment by andymc12** [comment]
#build

**Pull request merged**